app/lib/nerves_hub.rb

```diff
@@ -0,0 +1,270 @@
+    # in the CA store doesn't work for some reason?
+    @conn.ca_file    = self.current_ca_file
+    # Don't think this is absolutely needed.
+    @conn.cert_store = nil
+    @conn            = obj
+    @conn.use_ssl    = true
+    @conn.cert       = current_cert
+    @conn.key        = current_key
+    @conn
+  end
+
+  # HTTP connection.
+  def self.conn
+    active? && !@conn ? set_conn : @conn
+  end
+
+private
+
+  # Helper for making requests to a device url on NervesHub
+  def self.devices_path(*chunks)
+    ["/orgs", NERVES_HUB_ORG, "devices"].concat(chunks).join("/")
+  end
+
+  # Helper for making requests to a particular device on NervesHub
+  def self.device_path(serial_number)
+    devices_path(serial_number)
+  end
+
+  # Helper for making signing requests for a device on NervesHub
+  def self.device_sign_path(serial_number)
+    devices_path(serial_number, "certificates", "sign")
+  end
+
+  # Generates a key on behalf of a NervesHub device
+  def self.generate_device_key
+    OpenSSL::PKey::EC.new("prime256v1").generate_key!
+  end
+
+  # Generates a CSR on behalf of a NervesHub device
+  # This CSR is POSTed to the NervesHub API.
+  # The signed key and a cert are then returned by NervesHub.
+  # This key and cert must be passed to a NervesHub device. In the case of
+  # FarmBotOS, it will be delivered via AMQP.
+  def self.generate_device_csr(serial_number, key)
+    options = {
+      :organization => NERVES_HUB_ORG,
+    }
+
+    request = OpenSSL::X509::Request.new
+    request.version = 0
+    request.subject = OpenSSL::X509::Name.new([
+      ['O', options[:organization], OpenSSL::ASN1::UTF8STRING],
+    ])
+    request.public_key = real_public_key(key)
+    request.sign(key, OpenSSL::Digest::SHA1.new)
+  end
+
+  # This is because OpenSSL::PKey::EC doesn't follow the same API
+  # as other private keys for some reason.
+  # https://github.com/ruby/openssl/issues/29#issuecomment-149799052
+  def self.real_public_key(k)
+    point = k.public_key
+    pub = OpenSSL::PKey::EC.new(point.group)
+    pub.public_key = point
+    pub
+  end
+
+  # Cert for authenticating Farmbot API (NOT FARMBOT OS) to NervesHub
+  def self.try_env_cert
+    OpenSSL::X509::Certificate.new(ENV['NERVES_HUB_CERT']) if ENV['NERVES_HUB_CERT']
+  end
+
+  # Cert for authenticating Farmbot API (NOT FARMBOT OS) to NervesHub
+  def self.try_file_cert
+    OpenSSL::X509::Certificate.new(File.read(NERVES_HUB_CERT_PATH)) if File.exist?(NERVES_HUB_CERT_PATH)
+  end
+
+  # Cert for authenticating Farmbot API (NOT FARMBOT OS) to NervesHub
+  def self.try_env_key
+    OpenSSL::PKey::EC.new(ENV['NERVES_HUB_KEY']) if ENV['NERVES_HUB_KEY']
+  end
+
+  # Private Key for authenticating Farmbot API (NOT FARMBOT OS) to NervesHub
+  def self.try_file_key
+    OpenSSL::PKey::EC.new(File.read(NERVES_HUB_KEY_PATH)) if File.exist?(NERVES_HUB_KEY_PATH)
+  end
+
+  # Private Key for authenticating Farmbot API (NOT FARMBOT OS) to NervesHub
+  def self.try_env_ca_file
```

Comment:
I got this method, comment and the following method wrong. Sorry.

It should be `self.try_file_ca_file` and the next method should be `self.try_env_ca_file`

The comment should also say something like `CA file for authenticating Farmbot API (NOT FARMBOT OS) to NervsHub